scripts/installer

```diff
@@ -0,0 +1,383 @@
+subjects:
+  - kind: ServiceAccount
+    name: tekton-dashboard
+    namespace: $INSTALL_NAMESPACE
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-dashboard-pipelines
+EOF
+
+cat <<EOF >> $TMP_FILE
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: tekton-dashboard-triggers
+  namespace: $TRIGGERS_NAMESPACE
+  labels:
+    app.kubernetes.io/component: dashboard
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-dashboard
+subjects:
+  - kind: ServiceAccount
+    name: tekton-dashboard
+    namespace: $INSTALL_NAMESPACE
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-dashboard-triggers
+EOF
+}
+
+# install invokes kubectl apply with the downloaded manifest.
+install() {
+  kubectl create ns $INSTALL_NAMESPACE || true
+  kubectl apply -f $TMP_FILE
+}
+
+# uninstall invokes kubectl delete with the downloaded manifest.
+uninstall() {
+  kubectl delete -f $TMP_FILE
+}
+
+# uninstall invokes kubectl delete with the downloaded manifest.
+resolve() {
+  cat $TMP_FILE
+}
+
+# fail_trap is executed if an error occurs.
+fail_trap() {
+  result=$?
+  cleanup
+  exit $result
+}
+
+# help provides possible cli installation arguments
+help () {
+  echo -e "Global command syntax:"
+  echo -e "\tinstaller COMMAND [OPTIONS]"
+  echo -e ""
+  echo -e "Accepted commands:"
+  echo -e "\thelp|h\t\t\t\t\tPrints this help"
+  echo -e "\tinstall|i\t\t\t\tInstalls the dashboard"
+  echo -e "\tuninstall|u\t\t\t\tUninstalls the dashboard"
+  echo -e "\tbuild|b\t\t\t\t\tBuilds the manifests and docker image and outputs manfiests in the console"
+  echo -e ""
+  echo -e "Accepted options:"
+  echo -e "\t[--debug]\t\t\t\tPrints additional messages in the console"
+  echo -e "\t[--no-csrf-secure-cookie]\t\tAllows unsecure CSRF cookie"
+  echo -e "\t[--openshift]\t\t\t\tWill build manifests for openshift"
+  echo -e "\t[--readonly]\t\t\t\tWill build manifests for a readonly deployment"
+  echo -e "\t[--namespace <namespace>]\t\tWill override install namespace"
+  echo -e "\t[--pipelines-namespace <namespace>]\tOverride the namespace where Tekton Pipelines is installed (defaults to tekton-pipelines)"
+  echo -e "\t[--triggers-namespace <namespace>]\tOverride the namespace where Tekton Triggers is installed (defaults to tekton-pipelines)"
+  echo -e "\t[--ingress-url <url>]\t\t\tWill create an additional ingress with the specified url"
+  echo -e "\t[--ingress-secret <secret>]\t\tWill add ssl support to the ingress"
+}
+
+# cleanup temporary files
+cleanup() {
+  if [[ -d "${TMP_ROOT:-}" ]]; then
+    rm -rf "$TMP_ROOT"
+  fi
+}
+
+# Execution
+
+#Stop execution on any error
+trap "fail_trap" EXIT
+
+set -e
+set -u
+
+# Parsing command
+case $1 in
+  'help'|h)
+    help
+    exit 0
+    ;;
+  'install'|i)
+    ACTION="install"
+    ;;
+  'uninstall'|u)
+    ACTION="uninstall"
+    ;;
+  'build'|b)
+    ACTION="build"
+    ;;
+  *)
```

Comment:
we should also fallback to this if no command provided, e.g. `./scripts/installer`
currently get an error: `./scripts/installer: line 290: $1: unbound variable`

Reply:
fixed